vendor/k8s.io/kubernetes/cmd/libs/go2idl/client-gen/generators/client_generator.go

```diff
@@ -40,8 +40,31 @@ func NameSystems() namer.NameSystems {
 
 		"SecurityContextConstraints": "SecurityContextConstraints",
 	}
+
+	publicNamer := &ExceptionNamer{
+		Exceptions: map[string]string{
+			// these exceptions are used to deconflict the generated code
+			"github.com/openshift/origin/pkg/build/api/v1.Build":       "BuildResource",
+			"github.com/openshift/origin/pkg/build/api.Build":          "BuildResource",
+			"github.com/openshift/origin/pkg/image/api/v1.Image":       "ImageResource",
+			"github.com/openshift/origin/pkg/image/api.Image":          "ImageResource",
+			"github.com/openshift/origin/pkg/project/api/v1.Project":   "ProjectResource",
+			"github.com/openshift/origin/pkg/project/api.Project":      "ProjectResource",
+			"github.com/openshift/origin/pkg/route/api/v1.Route":       "RouteResource",
+			"github.com/openshift/origin/pkg/route/api.Route":          "RouteResource",
+			"github.com/openshift/origin/pkg/template/api/v1.Template": "TemplateResource",
+			"github.com/openshift/origin/pkg/template/api.Template":    "TemplateResource",
+			"github.com/openshift/origin/pkg/user/api/v1.User":         "UserResource",
+			"github.com/openshift/origin/pkg/user/api.User":            "UserResource",
```

Comment:
> We could derive the exceptions from GroupName==ResourceName. This would avoid that we do an UPSTREAM commit for each new group (but how often does that happen).

This only happens for stuttery resources like `route.route.openshift.io`.  An upstream fix would be better since a code generator shouldn't fail on it.